src/components/GalleryCreate/index.tsx

```diff
@@ -0,0 +1,227 @@
+import React, { useState, useEffect } from 'react';
+// import { store } from 'apis/firebase';
+import styled from 'styled-components';
+import ReactModal from 'react-modal';
+import closeButton from '../../assets/icons/closeButton.svg';
+
+function GalleryRead() {
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const [imgFile, setImgFile] = useState<File | null>();
+  const [preview, setPreview] = useState<string | null>('');
+
+  const openModal = () => {
+    setModalOpen(true);
+  };
+
+  const closeModal = () => {
+    setModalOpen(false);
+    setPreview(null);
+  };
+
+  const onChangeImg = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files !== null) {
+      const file = event.target.files[0];
+      setImgFile(file);
+    } else {
+      setImgFile(null);
+    }
+  };
+
+  useEffect(() => {
+    if (imgFile) {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        setPreview(reader.result as string);
+      };
+      reader.readAsDataURL(imgFile);
+    } else {
+      setPreview(null);
+    }
+  }, [imgFile]);
+
+  const imgRegistration = () => {
+    if (preview === null) {
+      alert('사진 업로드 후 등록해주세요');
+    } else {
+      closeModal();
+    }
+  };
```

Comment:
아래와 같이 if문의 state는 trusy, falsy로 간단하게 표현하고,  else보단 early return을 사용하시는 건
어떨까요!?
ref) https://velog.io/@jangws/9.-Else-if%EC%99%80-Else%EB%A5%BC-%ED%94%BC%ED%95%98%EA%B3%A0-Early-Return%ED%95%98%EC%9E%90
```suggestion
  const imgRegistration = () => {
    if (preview) {
      alert('사진 업로드 후 등록해주세요');
      return;
    } 
      closeModal();
  };
```

Reply:
오!! 코드가 훨씬 깔끔해지는거 같네요!
Early return에 대해 배워갑니다 🙇🏻‍♀️